src/runtime/providers/cloudflare.ts

```diff
@@ -0,0 +1,48 @@
+// https://developers.cloudflare.com/images/url-format
+
+import defu from 'defu'
+import { ProviderGetImage } from 'src'
+import { joinURL, encodeQueryItem, encodePath } from 'ufo'
+import { createOperationsGenerator } from '~image'
+
+const operationsGenerator = createOperationsGenerator({
+  keyMap: {
+    width: 'w',
+    height: 'h',
+    dpr: 'dpr',
+    fit: 'fit',
+    gravity: 'g',
+    quality: 'q',
+    format: 'f',
+    sharpen: 'sharpen'
+  },
+  valueMap: {
+    fit: {
+      cover: 'cover',
+      contain: 'contain',
+      fill: 'scale-down',
+      outside: 'crop',
+      inside: 'pad'
+    },
+    gravity: {
+      auto: 'auto',
+      side: 'side'
+    }
+  },
+  joinWith: ',',
+  formatter: (key, val) => encodeQueryItem(key, val)
+})
+
+const defaultModifiers = {
+  format: 'auto',
+  quality: 'auto'
+}
+
+export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL = '/' } = {}) => {
+  const mergeModifiers = defu(modifiers, defaultModifiers)
```

Comment:
Since it is a flat assignment, maybe we can simply merge objects with `{ ...defaultModifiers, ...modifiers }`